src/server/pfs/cmds/cmds.go

```diff
@@ -1144,6 +1145,26 @@ $ pachctl diff-file foo master path1 bar master path2
 	}
 	unmount.Flags().BoolVarP(&all, "all", "a", false, "unmount all pfs mounts")
 
+	var port uint16
+	s3gateway := &cobra.Command{
+		Use:   "s3gateway",
+		Short: "Expose pfs via an S3-like API. This command blocks.",
+		Long:  "Expose pfs via an S3-like API. This command blocks.",
+		Run: cmdutil.RunFixedArgs(0, func(args []string) error {
+			client, err := client.NewOnUserMachine(!*noMetrics, !*noPortForwarding, "user")
+			if err != nil {
+				return err
+			}
+			defer client.Close()
+			server := s3.Server(client, port)
+			if err := server.ListenAndServe(); err != http.ErrServerClosed {
+				return err
+			}
+			return nil
+		}),
+	}
+	s3gateway.Flags().Uint16VarP(&port, "port", "p", 30600, "The local port to bind the S3 gateway to.")
+
```

Comment:
Being able to start the server with `pachctl` seems pretty handy, but I think a more important access pattern is having the server get started in `pachd`'s main function and add that port to the pachd service. That way other pods scheduled on kubernetes will be able to talk to it to for their workloads. It's also the first step to exposing an s3 api to worker pods since they have a pachd instance running in the sidecar.

Reply:
I'll be adding that next!